cardano-launcher/src/Cardano/Shell/Environment.hs

```diff
@@ -0,0 +1,69 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Cardano.Shell.Environment
+  ( SubstitutionError(..)
+  , substituteEnvVars
+  , getLauncherOption'
+  ) where
+
+import           Cardano.Prelude
+
+import           Cardano.Shell.Launcher (LauncherOptions)
+import           Cardano.Shell.Template (substituteA)
+import           Data.Aeson (Result (..), Value (..), fromJSON)
+import           Data.Yaml (decodeFileEither)
+import           System.Environment (lookupEnv, setEnv)
+
+-- This is a example of how you can decode yaml and perform env var substitution
+-- | Parse `yaml` file, perform env var substitution, return @LauncherOptions@
+-- if successful
+getLauncherOption' :: FilePath -> IO (Either SubstitutionError LauncherOptions)
+getLauncherOption' yamlPath = do
+  setEnv "XDG_DATA_HOME" "Foo"
+  setEnv "DAEDALUS_CONFIG" "Bar"
+  eSubstituted <- runExceptT $ do
+    decodedVal <- withExceptT (\e -> FailedToDecodeFile (show e)) $ 
+      ExceptT $ decodeFileEither yamlPath
+    substituted <- substituteEnvVars decodedVal
+    return substituted
+  case eSubstituted of
+    Left err -> return . Left $ err
+    Right sub -> case fromJSON sub of
+        Error _ -> return . Left $ FailedToParseLauncherOption
+        Success decoded -> return . Right $ decoded
```

Comment:
This is just an example code I wrote to test that it actually works.
After this and https://github.com/input-output-hk/cardano-shell/pull/275 has been merged. I'll work on substituting the values the launcher actually needs